indico/modules/events/registration/controllers/api/misc.py

```diff
@@ -5,10 +5,20 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
+from operator import itemgetter
+
 from indico.modules.events.controllers.base import RHProtectedEventBase
+from indico.util.iterables import group_list
 
 
 class RHAPIRegistrationForms(RHProtectedEventBase):
     def _process(self):
         from indico.modules.events.registration.schemas import RegistrationFormPrincipalSchema
         return RegistrationFormPrincipalSchema(many=True).jsonify(self.event.registration_forms)
+
+
+class RHAPIEventSessionBlocks(RHProtectedEventBase):
+    def _process(self):
+        schema = [{'id': sb.id, 'start_dt': sb.start_dt.astimezone(self.event.tzinfo).date().isoformat(),
+                   'full_title': sb.full_title} for s in self.event.sessions for sb in s.blocks]
```

Comment:
I think this needs an `if s.can_access(session.user)` check to skip sessions which the user cannot access (some events have both public and restricted sessions).